# Exploring eBay Car Sales Data

The aim of this project is to clean the data and analyze the included used car listings.


## **Imports and reading data**

In [5]:
import pandas as pd

autos = pd.read_csv('Exploring eBay Car Sales Data/autos.csv', encoding='Latin-1')
autos

autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

There are 20 columns, only 5 of them has numerical values.

There are also 5 columns that have null values, these are:


*   vehicleType
*   gearbox
*   model
*   fuelType
*   notRepairedDamage

This dataset takes 7.6 MB of memory.




## **Data cleaning**

In [6]:
autos.rename({'yearOfRegistration':'registration_year',
              'monthOfRegistration':'registration_month',
              'notRepairedDamage':'unrepaired_damage',
              'dateCreated':'ad_created'}, axis=1, inplace=True)
autos.dtypes

dateCrawled           object
name                  object
seller                object
offerType             object
price                 object
abtest                object
vehicleType           object
registration_year      int64
gearbox               object
powerPS                int64
model                 object
odometer              object
registration_month     int64
fuelType              object
brand                 object
unrepaired_damage     object
ad_created            object
nrOfPictures           int64
postalCode             int64
lastSeen              object
dtype: object

In [7]:
autos.describe(include='all')


dateCrawled         name  seller offerType  price abtest  \
count                 50000        50000   50000     50000  50000  50000   
unique                48213        38754       2         2   2357      2   
top     2016-04-02 11:37:04  Ford_Fiesta  privat   Angebot     $0   test   
freq                      3           78   49999     49999   1421  25756   
mean                    NaN          NaN     NaN       NaN    NaN    NaN   
std                     NaN          NaN     NaN       NaN    NaN    NaN   
min                     NaN          NaN     NaN       NaN    NaN    NaN   
25%                     NaN          NaN     NaN       NaN    NaN    NaN   
50%                     NaN          NaN     NaN       NaN    NaN    NaN   
75%                     NaN          NaN     NaN       NaN    NaN    NaN   
max                     NaN          NaN     NaN       NaN    NaN    NaN   

       vehicleType  registration_year  gearbox       powerPS  model  \
count        44905       50000.000000    47320  50000.000000  47242   
unique           8                NaN        2           NaN    245   
top      limousine                NaN  manuell           NaN   golf   
freq         12859                NaN    36993           NaN   4024   
mean           NaN        2005.073280      NaN    116.355920    NaN   
std            NaN         105.712813      NaN    209.216627    NaN   
min            NaN        1000.000000      NaN      0.000000    NaN   
25%            NaN        1999.000000      NaN     70.000000    NaN   
50%            NaN        2003.000000      NaN    105.000000    NaN   
75%            NaN        2008.000000      NaN    150.000000    NaN   
max            NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuelType       brand unrepaired_damage  \
count       50000        50000.000000    45518       50000             40171   
unique         13                 NaN        7          40                 2   
top     150,000km                 NaN   benzin  volkswagen              nein   
freq        32424                 NaN    30107       10687             35232   
mean          NaN            5.723360      NaN         NaN               NaN   
std           NaN            3.711984      NaN         NaN               NaN   
min           NaN            0.000000      NaN         NaN               NaN   
25%           NaN            3.000000      NaN         NaN               NaN   
50%           NaN            6.000000      NaN         NaN               NaN   
75%           NaN            9.000000      NaN         NaN               NaN   
max           NaN           12.000000      NaN         NaN               NaN   

                 ad_created  nrOfPictures    postalCode             lastSeen  
count                 50000       50000.0  50000.000000                50000  
unique                   76           NaN           NaN                39481  
top     2016-04-03 00:00:00           NaN           NaN  2016-04-07 06:17:27  
freq                   1946           NaN           NaN                    8  
mean                    NaN           0.0  50813.627300                  NaN  
std                     NaN           0.0  25779.747957                  NaN  
min                     NaN           0.0   1067.000000                  NaN  
25%                     NaN           0.0  30451.000000                  NaN  
50%                     NaN           0.0  49577.000000                  NaN  
75%                     NaN           0.0  71540.000000                  NaN  
max                     NaN           0.0  99998.000000                  NaN

*  **registration_month** and postal cods columns need to be investigated as they are treated as numerical but should be a category type
*   **nrOfPictures** also need to be closely investigated as it is numerical values but all of descriptive statistics sugests that it has all values as 0. It may be dropped.
*   Also when it comes to **powerPS**, the min value of 0 is illogical as no car can have 0 of power in PS. But the highest value of 177000 should also be checked as it is just suspiecious - unless it is mayba a plane and not car.
*  **Registration year** also suggests that min and max values are somehow wrong as in 1000 there were no cars and 9999 year is yet to come.
*   The **odometer** column should be a numerical type yet it needs to be formated as there are ',' and 'km'. The same for **price** columns.



In [8]:
autos['price'] = autos['price'].str.replace('$', '')
autos['price'] = autos['price'].str.replace(',', '')
autos['price'] = autos['price'].str.replace('.', '')
autos['price'] = autos['price'].astype(float)


In [9]:
autos['odometer'] = autos['odometer'].str.replace('km', '')
autos['odometer'] = autos['odometer'].str.replace(',', '')
autos['odometer'] = autos['odometer'].astype(float)


In [10]:
autos.rename({'odometer':'odometer_km'}, axis=1, inplace=True)

autos.describe(include='all')

dateCrawled         name  seller offerType         price  \
count                 50000        50000   50000     50000  5.000000e+04   
unique                48213        38754       2         2           NaN   
top     2016-04-02 11:37:04  Ford_Fiesta  privat   Angebot           NaN   
freq                      3           78   49999     49999           NaN   
mean                    NaN          NaN     NaN       NaN  9.840044e+03   
std                     NaN          NaN     NaN       NaN  4.811044e+05   
min                     NaN          NaN     NaN       NaN  0.000000e+00   
25%                     NaN          NaN     NaN       NaN  1.100000e+03   
50%                     NaN          NaN     NaN       NaN  2.950000e+03   
75%                     NaN          NaN     NaN       NaN  7.200000e+03   
max                     NaN          NaN     NaN       NaN  1.000000e+08   

       abtest vehicleType  registration_year  gearbox       powerPS  model  \
count   50000       44905       50000.000000    47320  50000.000000  47242   
unique      2           8                NaN        2           NaN    245   
top      test   limousine                NaN  manuell           NaN   golf   
freq    25756       12859                NaN    36993           NaN   4024   
mean      NaN         NaN        2005.073280      NaN    116.355920    NaN   
std       NaN         NaN         105.712813      NaN    209.216627    NaN   
min       NaN         NaN        1000.000000      NaN      0.000000    NaN   
25%       NaN         NaN        1999.000000      NaN     70.000000    NaN   
50%       NaN         NaN        2003.000000      NaN    105.000000    NaN   
75%       NaN         NaN        2008.000000      NaN    150.000000    NaN   
max       NaN         NaN        9999.000000      NaN  17700.000000    NaN   

          odometer_km  registration_month fuelType       brand  \
count    50000.000000        50000.000000    45518       50000   
unique            NaN                 NaN        7          40   
top               NaN                 NaN   benzin  volkswagen   
freq              NaN                 NaN    30107       10687   
mean    125732.700000            5.723360      NaN         NaN   
std      40042.211706            3.711984      NaN         NaN   
min       5000.000000            0.000000      NaN         NaN   
25%     125000.000000            3.000000      NaN         NaN   
50%     150000.000000            6.000000      NaN         NaN   
75%     150000.000000            9.000000      NaN         NaN   
max     150000.000000           12.000000      NaN         NaN   

       unrepaired_damage           ad_created  nrOfPictures    postalCode  \
count              40171                50000       50000.0  50000.000000   
unique                 2                   76           NaN           NaN   
top                 nein  2016-04-03 00:00:00           NaN           NaN   
freq               35232                 1946           NaN           NaN   
mean                 NaN                  NaN           0.0  50813.627300   
std                  NaN                  NaN           0.0  25779.747957   
min                  NaN                  NaN           0.0   1067.000000   
25%                  NaN                  NaN           0.0  30451.000000   
50%                  NaN                  NaN           0.0  49577.000000   
75%                  NaN                  NaN           0.0  71540.000000   
max                  NaN                  NaN           0.0  99998.000000   

                   lastSeen  
count                 50000  
unique                39481  
top     2016-04-07 06:17:27  
freq                      8  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

In [11]:
print(f"Shape of unique values for price: {autos.price.unique().shape}")
print(autos.price.describe())

Shape of unique values for price: (2357,)
count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64


In [12]:
autos.price.value_counts()

price
0.0        1421
500.0       781
1500.0      734
2500.0      643
1000.0      639
           ... 
414.0         1
79933.0       1
5198.0        1
18890.0       1
16995.0       1
Name: count, Length: 2357, dtype: int64

In [13]:
autos_cleaned = autos[autos['price'] > 0.0]
autos_cleaned.reset_index(drop=True, inplace=True)
autos_cleaned

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
48574  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
48575  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
48576  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
48577  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
48578  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offerType    price   abtest vehicleType  registration_year  \
0      privat   Angebot   5000.0  control         bus               2004   
1      privat   Angebot   8500.0  control   limousine               1997   
2      privat   Angebot   8990.0     test   limousine               2009   
3      privat   Angebot   4350.0  control  kleinwagen               2007   
4      privat   Angebot   1350.0     test       kombi               2003   
...       ...       ...      ...      ...         ...                ...   
48574  privat   Angebot  24900.0  control   limousine               2011   
48575  privat   Angebot   1980.0  control      cabrio               1996   
48576  privat   Angebot  13200.0     test      cabrio               2014   
48577  privat   Angebot  22900.0  control       kombi               2013   
48578  privat   Angebot   1250.0  control   limousine               1996   

         gearbox  powerPS   model  odometer_km  registration_month fuelType  \
0        manuell      158  andere     150000.0                   3      lpg   
1      automatik      286     7er     150000.0                   6   benzin   
2        manuell      102    golf      70000.0                   7   benzin   
3      automatik       71  fortwo      70000.0                   6   benzin   
4        manuell        0   focus     150000.0                   7   benzin   
...          ...      ...     ...          ...                 ...      ...   
48574  automatik      239      q5     100000.0                   1   diesel   
48575    manuell       75   astra     150000.0                   5   benzin   
48576  automatik       69     500       5000.0                  11   benzin   
48577    manuell      150      a3      40000.0                  11   diesel   
48578    manuell      101  vectra     150000.0                   1   benzin   

            brand unrepaired_damage           ad_created  nrOfPictures  \
0         peugeot              nein  2016-03-26 00:00:00             0   
1             bmw              nein  2016-04-04 00:00:00             0   
2      volkswagen              nein  2016-03-26 00:00:00             0   
3           smart              nein  2016-03-12 00:00:00             0   
4            ford              nein  2016-04-01 00:00:00             0   
...           ...               ...                  ...           ...   
48574        audi              nein  2016-03-27 00:00:00             0   
48575        opel              nein  2016-03-28 00:00:00             0   
48576        fiat              nein  2016-04-02 00:00:00             0   
48577        audi              nein  2016-03-08 00:00:00             0   
48578        opel              nein  2016-03-13 00:00:00             0   

       postalCode             lastSeen  
0           79588  2016-04-06 06:45:54  
1           71034  2016-04-06 14:45:08  
2           35394  2016-04-06 20:15:37  
3           33729  2016-03-15 03:16:28  
4           39218  2016-04-01 14:38:50  
...           ...                  ...  
4857

In [14]:
print(f"Shape of unique values for odometer_km: {autos_cleaned.odometer_km.unique().shape}")
print(autos_cleaned.odometer_km.describe())

Shape of unique values for odometer_km: (13,)
count     48579.000000
mean     125766.174685
std       39792.936148
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64


In [15]:

autos_cleaned.head()


dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  registration_year  \
0  privat   Angebot  5000.0  control         bus               2004   
1  privat   Angebot  8500.0  control   limousine               1997   
2  privat   Angebot  8990.0     test   limousine               2009   
3  privat   Angebot  4350.0  control  kleinwagen               2007   
4  privat   Angebot  1350.0     test       kombi               2003   

     gearbox  powerPS   model  odometer_km  registration_month fuelType  \
0    manuell      158  andere     150000.0                   3      lpg   
1  automatik      286     7er     150000.0                   6   benzin   
2    manuell      102    golf      70000.0                   7   benzin   
3  automatik       71  fortwo      70000.0                   6   benzin   
4    manuell        0   focus     150000.0                   7   benzin   

        brand unrepaired_damage           ad_created  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

There is no outliers when it comes to odometer_km column after cleaning the price column values. But from now on, there will be autos_cleaned set used as base set of data.


In [16]:
autos_cleaned['dateCrawled'] = autos_cleaned['dateCrawled'].str[:10]
autos_cleaned['ad_created'] = autos_cleaned['ad_created'].str[:10]
autos_cleaned['lastSeen'] = autos_cleaned['lastSeen'].str[:10]
autos_cleaned


/var/folders/nm/_nrzqcrn7tl7q4j1yjdtc8_c0000gn/T/ipykernel_44999/4157140256.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  autos_cleaned['dateCrawled'] = autos_cleaned['dateCrawled'].str[:10]
/var/folders/nm/_nrzqcrn7tl7q4j1yjdtc8_c0000gn/T/ipykernel_44999/4157140256.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  autos_cleaned['ad_created'] = autos_cleaned['ad_created'].str[:10]
/var/folders/nm/_nrzqcrn7tl7q4j1yjdtc8_c0000gn/T/ipykernel_44999/4157140256.py:3: SettingWithCopyWarning: 
A value is tr

dateCrawled                                               name  seller  \
0      2016-03-26                   Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1      2016-04-04         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   
2      2016-03-26                         Volkswagen_Golf_1.6_United  privat   
3      2016-03-12  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...  privat   
4      2016-04-01  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...  privat   
...           ...                                                ...     ...   
48574  2016-03-27   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon  privat   
48575  2016-03-28  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...  privat   
48576  2016-04-02                     Fiat_500_C_1.2_Dualogic_Lounge  privat   
48577  2016-03-08                 Audi_A3_2.0_TDI_Sportback_Ambition  privat   
48578  2016-03-14                                Opel_Vectra_1.6_16V  privat   

      offerType    price   abtest vehicleType  registration_year    gearbox  \
0       Angebot   5000.0  control         bus               2004    manuell   
1       Angebot   8500.0  control   limousine               1997  automatik   
2       Angebot   8990.0     test   limousine               2009    manuell   
3       Angebot   4350.0  control  kleinwagen               2007  automatik   
4       Angebot   1350.0     test       kombi               2003    manuell   
...         ...      ...      ...         ...                ...        ...   
48574   Angebot  24900.0  control   limousine               2011  automatik   
48575   Angebot   1980.0  control      cabrio               1996    manuell   
48576   Angebot  13200.0     test      cabrio               2014  automatik   
48577   Angebot  22900.0  control       kombi               2013    manuell   
48578   Angebot   1250.0  control   limousine               1996    manuell   

       powerPS   model  odometer_km  registration_month fuelType       brand  \
0          158  andere     150000.0                   3      lpg     peugeot   
1          286     7er     150000.0                   6   benzin         bmw   
2          102    golf      70000.0                   7   benzin  volkswagen   
3           71  fortwo      70000.0                   6   benzin       smart   
4            0   focus     150000.0                   7   benzin        ford   
...        ...     ...          ...                 ...      ...         ...   
48574      239      q5     100000.0                   1   diesel        audi   
48575       75   astra     150000.0                   5   benzin        opel   
48576       69     500       5000.0                  11   benzin        fiat   
48577      150      a3      40000.0                  11   diesel        audi   
48578      101  vectra     150000.0                   1   benzin        opel   

      unrepaired_damage  ad_created  nrOfPictures  postalCode    lastSeen  
0                  nein  2016-03-26             0       79588  2016-04-06  
1                  nein  2016-04-04             0       71034  2016-04-06  
2                  nein  2016-03-26             0       35394  2016-04-06  
3                  nein  2016-03-12             0       33729  2016-03-15  
4                  nein  2016-04-01             0       39218  2016-04-01  
...                 ...         ...           ...         ...         ...  
48574              nein  2016-03-27             0       82131  2016-04-01  
48575              nein  2016-03-28             0       44807  2016-04-02  
48576              nein  2016-04-02             0       73430  2016-04-04  
48577              nein  2016-03-08             0       35683  2016-04-05  
48578              nein  2016-03-13             0       45897  2016-04-06  

[48579 rows x 20 columns]

In [17]:
# dateCrawled
print(f"Normalized dateCrawled set:\n{autos_cleaned.dateCrawled.value_counts(normalize=True, dropna=False)} \n\n")

# ad_created
print(f"Normalized ad_created set:\n{autos_cleaned.ad_created.value_counts(normalize=True, dropna=False)}\\n\n")

# lastSeen
print(f"Normalized lastSeen set:\n{autos_cleaned.lastSeen.value_counts(normalize=True, dropna=False)}\n")


Normalized dateCrawled set:
dateCrawled
2016-04-03    0.038597
2016-03-20    0.037876
2016-03-21    0.037403
2016-03-12    0.036930
2016-03-14    0.036538
2016-04-04    0.036518
2016-03-07    0.036003
2016-04-02    0.035468
2016-03-28    0.034850
2016-03-19    0.034768
2016-03-15    0.034274
2016-03-29    0.034130
2016-03-30    0.033677
2016-04-01    0.033677
2016-03-08    0.033327
2016-03-09    0.033101
2016-03-22    0.033018
2016-03-11    0.032566
2016-03-23    0.032216
2016-03-26    0.032195
2016-03-10    0.032174
2016-03-31    0.031845
2016-03-17    0.031639
2016-03-25    0.031598
2016-03-27    0.031083
2016-03-16    0.029601
2016-03-24    0.029334
2016-03-05    0.025320
2016-03-13    0.015665
2016-03-06    0.014039
2016-04-05    0.013092
2016-03-18    0.012907
2016-04-06    0.003170
2016-04-07    0.001400
Name: proportion, dtype: float64 


Normalized ad_created set:
ad_created
2016-04-03    0.038844
2016-03-20    0.037938
2016-03-21    0.037609
2016-04-04    0.036888
2016-03-12  

In [18]:
autos_cleaned.registration_year.describe()

count    48579.000000
mean      2004.753000
std         88.631663
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64


There is a large number of users inactive since april 2016 in days 6-7, as there is no later date, I would suggest that data ends at 7 april 2016. However crawled date suggest the regular distribution for whole set.

The registtration year columns needs to be cleaned more, as the max and min values are illogic. As the car was invented in 1886, we will drop all values with registration year before that, and as the datasets ends in 2016, registration year should not exceeds 2016.


In [19]:
autos_cleaned = autos_cleaned[autos_cleaned['registration_year'].between(1886, 2016)]
autos_cleaned.reset_index(drop=True, inplace=True)

autos_cleaned.registration_year.value_counts(normalize=True)

registration_year
2000    0.067590
2005    0.062879
1999    0.062086
2004    0.057889
2003    0.057803
          ...   
1939    0.000021
1948    0.000021
1938    0.000021
1953    0.000021
1943    0.000021
Name: proportion, Length: 78, dtype: float64


Distribution of registration year suggests that most of cars are registered in 2000, 2005 and 1999 year.


In [20]:
len(autos_cleaned['brand'].unique())

40

## **Main analysis**

In [21]:
by_brand = autos_cleaned['brand'].value_counts(normalize=True)
by_brand

brand
volkswagen        0.211274
bmw               0.110038
opel              0.107575
mercedes_benz     0.096460
audi              0.086544
ford              0.069925
renault           0.047138
peugeot           0.029833
fiat              0.025636
seat              0.018268
skoda             0.016405
nissan            0.015270
mazda             0.015184
smart             0.014156
citroen           0.014028
toyota            0.012700
hyundai           0.010023
sonstige_autos    0.009873
volvo             0.009145
mini              0.008759
mitsubishi        0.008224
honda             0.007838
kia               0.007067
alfa_romeo        0.006639
porsche           0.006125
suzuki            0.005932
chevrolet         0.005697
chrysler          0.003512
dacia             0.002634
daihatsu          0.002506
jeep              0.002270
subaru            0.002142
land_rover        0.002099
saab              0.001649
jaguar            0.001563
daewoo            0.001499
trabant           0.00

There are 40 brands in dataset. One of the mos popular are: volkswagen, bmw and opel.
Now we are going to calculate the mean price for those that have at least 1% of total.


In [22]:
names = by_brand[by_brand > 0.01]


brand_prices = {}
for i in list(names.index):
    brand_prices[i] = autos_cleaned[autos_cleaned['brand'] == i]['price'].mean()

brand_prices

{'volkswagen': 6729.81956411556,
 'bmw': 8571.480147917478,
 'opel': 5432.479195699781,
 'mercedes_benz': 30829.021092362345,
 'audi': 9336.687453600594,
 'ford': 7456.547932618683,
 'renault': 2474.8646069968195,
 'peugeot': 3094.0172290021537,
 'fiat': 2813.748538011696,
 'seat': 4397.230949589683,
 'skoda': 6368.0,
 'nissan': 4743.40252454418,
 'mazda': 4112.596614950635,
 'smart': 3580.2239031770046,
 'citroen': 45486.68549618321,
 'toyota': 5167.091062394604,
 'hyundai': 5365.254273504273}

The most expensive brands are citroen (45486 on avg) and mercedes_benz (with 30829 on avg).
The cheapest one are fiat(2813 and renault(2474).

In [25]:
brand_mileage = {}
for i in list(names.index):
    brand_mileage[i] = autos_cleaned[autos_cleaned['brand'] == i]['odometer_km'].mean()
brand_mileage

bp = pd.Series(brand_prices)
bm = pd.Series(brand_mileage)
df = pd.DataFrame(bp, columns=['mean_price'])
df['mean_mileage'] = bm
df

mean_price   mean_mileage
volkswagen      6729.819564  128713.634060
bmw             8571.480148  132575.905021
opel            5432.479196  129314.154888
mercedes_benz  30829.021092  130792.628774
audi            9336.687454  129157.386785
ford            7456.547933  124243.491577
renault         2474.864607  128071.331213
peugeot         3094.017229  127153.625269
fiat            2813.748538  117121.971596
seat            4397.230950  121131.301290
skoda           6368.000000  110848.563969
nissan          4743.402525  118330.995792
mazda           4112.596615  124464.033850
smart           3580.223903   99326.777610
citroen        45486.685496  119572.519084
toyota          5167.091062  115944.350759
hyundai         5365.254274  106442.307692


Najniższym średnim przebiegiem charakteryzuje się marka 'smart'. Najwyższym natomiast - 'bmw'.

Najniższa średnia cena obserwowana jest dla 'renault', a najwyższa dla 'citroen'.

Wyniki te można uzasadnić, jednak kategoryczne określenie powodów pozostaje jeszcze do przeanalizowania. Smart może mieć najniższy średni przebieg ze względu na to, że auta tej firmy są autami typowo miejskimi, rzadko wykorzystywanymi na przebywanie długich tras -stąd mniejszy średni przebieg. BMW natomiast wręcz przeciwnie, ponadto jest to marka ciesząca się dużą popularnością od wielu lat, stąd rynek może być wysycony starymi egzemplarzami, które przez lata użytkowania siłą rzeczy pokonały znaczną liczbę km.

W kwestii ceny jednak, nie przychodzi mi szczególne uzasadnienie tak dużej różnicy pomiedzy skrajnymi średnimi, wymaga to więc jeszcze lepszej znajmości danych i kolejnych analiz.